theoplayer/how-to-guides/01-ads/13-custom-ssai-integration.md

```diff
@@ -0,0 +1,158 @@
+# Custom Server Side Ad Insertion
+
+THEOplayer provides a way to integrate with third-party advertisement providers,
+and have them report their ads and ad-related events through the THEOplayer APIs.
+
+## Android SDK
+
+### Prerequisites
+
+Your THEOplayer SDK needs to be on version 7.6.0 or higher.
+
+### Integration ID
+
+To make use of the custom server side ad insertion, first you need an integration ID which will identify the provided sources and ads to the specific ad provider.
+
+### ServerSideAdIntegrationFactory
+
+You would also need to register a `ServerSideAdIntegrationFactory` that will receive a `ServerSideAdIntegrationController`.
+The controller can be used to create the THEOplayer `AdBreak`/`Ad` objects using the `AdBreakInit`/`AdInit` data.
+Additionally, it can be used to keep the player's state up-to-date such as when an ad begins or ends.
+
+The `ServerSideAdIntegrationFactory` requires an implementation of the `ServerSideAdIntegrationHandler` interface.
+The handler will provide few callbacks that can be overridden which allows you to setup and clear the integration as needed.
+
+For a concrete implementation of a YoSpace Connector can be found [here](https://github.com/THEOplayer/android-connector/blob/master/connectors/yospace/src/main/java/com/theoplayer/android/connector/yospace/internal/YospaceAdIntegration.kt).
+
+Meanwhile, an empty implementation would look like below. Details to follow.
+
+```kotlin
+player.ads.registerServerSideIntegration("integrationID", object : ServerSideAdIntegrationFactory {
```

Comment:
Nit-pick: I *think* this can be a closure?
```kotlin
player.ads.registerServerSideIntegration("integrationID") { controller ->
  return object : ServerSideAdIntegrationHandler { /* ... */ }
}